pkg/controller/sls/machineGroupBinding_controller.go

```diff
@@ -73,44 +71,16 @@ type machineGroupBindingConnector struct {
 func (c *machineGroupBindingConnector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
 	cr, ok := mg.(*aliv1alpha1.MachineGroupBinding)
 	if !ok {
-		return nil, errors.New(errNotMachineGroupBinding)
+		return nil, errors.New(errNotLogtail)
 	}
 
-	var (
-		sel    *xpv1.SecretKeySelector
-		region string
-	)
-
-	switch {
-	case cr.GetProviderConfigReference() != nil:
-		if err := c.usage.Track(ctx, mg); err != nil {
-			return nil, errors.Wrap(err, errTrackUsage)
-		}
-
-		pc := &v1alpha1.ProviderConfig{}
-		if err := c.client.Get(ctx, types.NamespacedName{Name: cr.Spec.ProviderConfigReference.Name}, pc); err != nil {
-			return nil, errors.Wrap(err, errGetProviderConfig)
-		}
-		if s := pc.Spec.Credentials.Source; s != xpv1.CredentialsSourceSecret {
-			return nil, errors.Errorf(errFmtUnsupportedCredSource, s)
-		}
-		sel = pc.Spec.Credentials.SecretRef
-		region = pc.Spec.Region
-	default:
-		return nil, errors.New(errNoProvider)
-	}
-
-	if sel == nil {
-		return nil, errors.New(errNoConnectionSecret)
-	}
-
-	s := &corev1.Secret{}
-	nn := types.NamespacedName{Namespace: sel.Namespace, Name: sel.Name}
-	if err := c.client.Get(ctx, nn, s); err != nil {
-		return nil, errors.Wrap(err, errGetConnectionSecret)
+	info, err := util.PrepareClient(ctx, mg, cr, c.client, c.usage, cr.Spec.ProviderConfigReference.Name)
```

Comment:
This was not introduced in this PR, but FWIW packages named `util` are considered an anti-pattern. See "bad package names" at https://blog.golang.org/package-names.

Reply:
Yeah. Fixing this bad package in https://github.com/crossplane/provider-alibaba/pull/96.